src/solver/expressions/include/antares/solver/expressions/visitors/NodeVisitor.h

```diff
@@ -45,37 +43,59 @@ class NodeVisitor
+        (
+          [&nodeDispatchFunctions] {
+              nodeDispatchFunctions[typeid(NodeTypes)] = &tryVisit<R,
+                                                                   NodeVisitor<R, Args...>,
+                                                                   NodeTypes>;
+          }(),
+          ...);
+        return nodeDispatchFunctions;
+    }
 
     R dispatch(const Node& node, Args... args)
     {
-        using FunctionT = std::optional<R> (*)(const Node&, NodeVisitor<R, Args...>&, Args... args);
-        static const std::vector<FunctionT> allNodeVisitList{
-          &tryVisit<R, NodeVisitor<R, Args...>, AddNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, SubtractionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, MultiplicationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, DivisionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, EqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LessThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, GreaterThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, NegationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ParameterNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, VariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LiteralNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, PortFieldNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentVariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentParameterNode>};
-        for (auto f: allNodeVisitList)
+        const static std::unordered_map<std::type_index, FunctionT>
+          nodeDispatchFunctions = NodesVisitList<AddNode,
+                                                 SubtractionNode,
+                                                 MultiplicationNode,
+                                                 DivisionNode,
+                                                 EqualNode,
+                                                 LessThanOrEqualNode,
+                                                 GreaterThanOrEqualNode,
+                                                 NegationNode,
+                                                 ParameterNode,
+                                                 VariableNode,
+                                                 LiteralNode,
+                                                 PortFieldNode,
+                                                 ComponentVariableNode,
+                                                 ComponentParameterNode>();
+        try
+        {
+            return nodeDispatchFunctions.at(typeid(node))(node, *this, args...);
+        }
+        catch (std::exception&)
         {
-            if (auto ret = f(node, *this, args...); ret.has_value())
-            {
-                return ret.value();
-            }
+            logs.error() << "Antares::Solver::Visitor: could not visit the node!";
```

Comment:
```suggestion
            logs.error() << "Antares::Solver::Visitor: could not visit the node! (exception " << ex.what() << ")";
```

https://en.cppreference.com/w/cpp/error/exception/what

Reply:
not necessary, when the exception is re-thrown;